main.go

```diff
@@ -147,6 +157,9 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	req, err := http.NewRequest("POST", url, bytes.NewBuffer(b))
+	if esUsername != "" {
+		req.Header.Add("Authorization", "Basic "+basicAuth(esUsername, esPassword))
```

Comment:
Won't this crash if esPassword isn't specified ? (can you test ?)

Reply:
I added one check to verify that part. Of course it's working if you don't want authentication.

Reply:
Apparently it doesn't crash because the default for a string is `""` and not `nil`.
Please revert to this version, ot was better.